Source/App.js

```diff
@@ -1 +1,417 @@
+        //Set the entity availability to the same interval as the simulation time.
+        availability : new Cesium.TimeIntervalCollection([new Cesium.TimeInterval({
+            start : viewer.clock.startTime,
+            stop : viewer.clock.stopTime
+        })]),
+        name : 'drone',
+        position : position,
+        orientation : new Cesium.VelocityOrientationProperty(position),
+        model : {
+            uri : './Source/SampleData/Models/CesiumDrone.gltf',
+            minimumPixelSize : 128,
+            maximumScale : 2000
+        },
+        //Show the path as a line sampled in 100 second increments.
+        path : {
+            resolution : 100,
+            material : new Cesium.PolylineGlowMaterialProperty({
+                glowPower : 0.1,
+                color : Cesium.Color.YELLOW
+            }),
+            width : 10
+        }
+    });
+
+    // Interpolate smoothly between position sample points.
+    drone.position.setInterpolationOptions({
+        interpolationDegree : 2,
+        interpolationAlgorithm : Cesium.HermitePolynomialApproximation
+    });
+
+    //////////////////////////////////////////////////////////////////////////
+    // Custom mouse interaction for highlighting and selecting
+    //////////////////////////////////////////////////////////////////////////
+
+    // If the mouse is over a point of interest, change the entity billboard scale and color
+    var previousPickedEntity = undefined;
+    var handler = new Cesium.ScreenSpaceEventHandler(viewer.scene.canvas);
+    handler.setInputAction(function(movement) {
+        var pickedPrimitive = viewer.scene.pick(movement.endPosition);
+        var pickedEntity = (Cesium.defined(pickedPrimitive)) ? pickedPrimitive.id : undefined;
+        // Unhighlight the previously picked entity
+        if (Cesium.defined(previousPickedEntity)) {
+            previousPickedEntity.billboard.scale = 1.0;
+            previousPickedEntity.billboard.color = Cesium.Color.WHITE;
+        }
+        // Highlight the currently picked entity
+        if (Cesium.defined(pickedEntity) && Cesium.defined(pickedEntity.billboard)) {
+            pickedEntity.billboard.scale = 2.0;
+            pickedEntity.billboard.color = Cesium.Color.ORANGERED;
+            previousPickedEntity = pickedEntity;
+        }
+    }, Cesium.ScreenSpaceEventType.MOUSE_MOVE);
+
+
+    //////////////////////////////////////////////////////////////////////////
+    // Setup Camera Modes
+    //////////////////////////////////////////////////////////////////////////
+
+    var freeModeElement = document.getElementById('freeMode');
+    var droneModeElement = document.getElementById('droneMode');
+
+    // Create a follow camera by using drone transformation to update camera
+    var scratch = new Cesium.Matrix4();
+    var camera =  viewer.scene.camera;
+    function followDrone() {
+        drone._getModelMatrix(viewer.clock.currentTime, scratch);
```

Comment:
We shouldn't teach users to access private functions and variables.  Is there another way to do this?  If not, maybe we should add one to Cesium